### PR TITLE
EventDb: WriteMarker

### LINKED ISSUE #768: Evenet db: WriteMarker table
We need an event database [WrtieMarker](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/models.go#L1018) table. this is populated during [read_redeem](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/blobber.go#L556).

We need code for the new table and add WrtieMarker method. There are four foreign keys into other tables to arrange.

We need to emit an add WrtieMarker event in [commitBlobberConnection](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/blobber.go#L348) which includes the input WrtieMarker sent by the blobber plus the transaction hash.

We will need to coordinate with blobber to include `LookUpHash`, `Name` and `ContentHash` into the Write marker. This is because Explorer requires these fields. https://github.com/0chain/blobber/issues/467

| WriteMarker            | type   | bobber.WriteMarker     | type      | notes                              |
|------------------------|--------|------------------------|-----------|------------------------------------|
| Client                 | uint   | ClinetId               | string    | foreign key into user table        |
| BlobberId              | uint   | BlobberId              | string    | foreign key into blobber table     |
| AllocationId           | uint   | AllocationId           | string    | foreign key into allocation table  |
| TransactionId          | uint   | t.Hash                 | string    | foreign key into transaction table |
| AllocationRoot         | string | AllocationRoot         | string    |                                    |
| PreviousAllocationRoot | string | PreviousAllocationRoot | string    |                                    |
| Size                   | int64  | Size                   | int64     |                                    |
| Timestamp              | int64  | Timestamp              | Timestamp |                                    |
| Signature              | string | Singature              | string    |                                    |
| BlockNumber             | int64  | txn.Round      | int64     |                                    |

### PR DESCRIPTION
**Pull request by ssardana96**
Tasks to complete before merging PR:
- [x]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [x]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [x]  Merge your system tests PR to master BEFORE merging this PR  
  
Associated PRs (Link as appropriate):
...

### CODE REVIEW

**Comment by ssardana96** [comment]
Table Description Created
```
events_db=> \d write_markers;
                                              Table "public.write_markers"
          Column          |           Type           | Collation | Nullable |                  Default
--------------------------+--------------------------+-----------+----------+-------------------------------------------
 id                       | bigint                   |           | not null | nextval('write_markers_id_seq'::regclass)
 created_at               | timestamp with time zone |           |          |
 updated_at               | timestamp with time zone |           |          |
 deleted_at               | timestamp with time zone |           |          |
 client_id                | text                     |           |          |
 blobber_id               | text                     |           |          |
 allocation_id            | text                     |           |          |
 transaction_id           | text                     |           |          |
 allocation_root          | text                     |           |          |
 previous_allocation_root | text                     |           |          |
 size                     | bigint                   |           |          |
 timestamp                | bigint                   |           |          |
 signature                | text                     |           |          |
 block_number             | bigint                   |           |          |
Indexes:
    "write_markers_pkey" PRIMARY KEY, btree (id)
    "idx_write_markers_deleted_at" btree (deleted_at)
Foreign-key constraints:
    "fk_blobbers_write_markers" FOREIGN KEY (blobber_id) REFERENCES blobbers(blobber_id)
```

**Comment by ssardana96** [comment]
write_markers table is getting populated when uploading file:
```
 id |          created_at           |          updated_at           | deleted_at |                            client_id
       |                            blobber_id                            |                          allocation_id
|                          transaction_id                          |                         allocation_root                          | previ
ous_allocation_root | size | timestamp  |                            signature                             | block_number
----+-------------------------------+-------------------------------+------------+-----------------------------------------------------------
-------+------------------------------------------------------------------+------------------------------------------------------------------
+------------------------------------------------------------------+------------------------------------------------------------------+------
--------------------+------+------------+------------------------------------------------------------------+--------------
  1 | 2022-01-06 15:07:17.595725+00 | 2022-01-06 15:07:17.619556+00 |            | 9984fb0d9647f57e8aa611a650c4d40fb34a4ad4fca204915fd64a4312
fd56d6 | f65af5d64000c7cd2883f4910eb69086f9d6e6635c744e62afcfab58b938ee25 | 70e72111daec0a10dcd9a9561a2fedeaa8e763c282b57d096a66b3b9b79006ce
| a92a9f23389f5df6dd05c8e202f47abb84a7226b0d37435aa39dcca31a89e546 | 4b4d2e082f7fccd115dc249b292eb5e4b7baaca65011c7abe06ca28d905e678a |
                    |  152 | 1641481633 | defa53bffdedfd2212c2cc4dc34aa5f58bb3bc39106d96620d113c8bf840c10a |         4183
  2 | 2022-01-06 15:07:21.302309+00 | 2022-01-06 15:07:21.307457+00 |            | 9984fb0d9647f57e8aa611a650c4d40fb34a4ad4fca204915fd64a4312
fd56d6 | 7a90e6790bcd3d78422d7a230390edc102870fe58c15472073922024985b1c7d | 70e72111daec0a10dcd9a9561a2fedeaa8e763c282b57d096a66b3b9b79006ce
| a52a24cb0823309c4b72a6ca4d58955cf5695dd38aced2181747136366034022 | 4b4d2e082f7fccd115dc249b292eb5e4b7baaca65011c7abe06ca28d905e678a |
                    |  152 | 1641481633 | 95b10e6d25b82b6e123d94cb93318de1fabded4bb1d3045447c2e9b0d3d0051c |         4194
```

**Comment by ssardana96** [comment]
I've created foreign key - blobberid, but rest are not there as table doesn't exists
Have created an issue for the same [859](https://github.com/0chain/0chain/issues/859)

**Comment by ssardana96** [comment]
New API /getMinerList response
http://localhost:7171/v1/screst/6dba10422e368813802877a85039d3985d96760ed844092319743fb3a76712d7/getWriteMarkers?allocation_id=70e72111daec0a10dcd9a9561a2fedeaa8e763c282b57d096a66b3b9b79006ce
```json
[
  {
    "ID": 1,
    "CreatedAt": "2022-01-06T15:07:17.595725Z",
    "UpdatedAt": "2022-01-06T15:07:17.619556Z",
    "DeletedAt": null,
    "client_id": "9984fb0d9647f57e8aa611a650c4d40fb34a4ad4fca204915fd64a4312fd56d6",
    "blobber_id": "f65af5d64000c7cd2883f4910eb69086f9d6e6635c744e62afcfab58b938ee25",
    "allocation_id": "70e72111daec0a10dcd9a9561a2fedeaa8e763c282b57d096a66b3b9b79006ce",
    "transaction_id": "a92a9f23389f5df6dd05c8e202f47abb84a7226b0d37435aa39dcca31a89e546",
    "allocation_root": "4b4d2e082f7fccd115dc249b292eb5e4b7baaca65011c7abe06ca28d905e678a",
    "previous_allocation_root": "",
    "size": 152,
    "timestamp": 1641481633,
    "signature": "defa53bffdedfd2212c2cc4dc34aa5f58bb3bc39106d96620d113c8bf840c10a",
    "block_number": 4183
  },
  {
    "ID": 2,
    "CreatedAt": "2022-01-06T15:07:21.302309Z",
    "UpdatedAt": "2022-01-06T15:07:21.307457Z",
    "DeletedAt": null,
    "client_id": "9984fb0d9647f57e8aa611a650c4d40fb34a4ad4fca204915fd64a4312fd56d6",
    "blobber_id": "7a90e6790bcd3d78422d7a230390edc102870fe58c15472073922024985b1c7d",
    "allocation_id": "70e72111daec0a10dcd9a9561a2fedeaa8e763c282b57d096a66b3b9b79006ce",
    "transaction_id": "a52a24cb0823309c4b72a6ca4d58955cf5695dd38aced2181747136366034022",
    "allocation_root": "4b4d2e082f7fccd115dc249b292eb5e4b7baaca65011c7abe06ca28d905e678a",
    "previous_allocation_root": "",
    "size": 152,
    "timestamp": 1641481633,
    "signature": "95b10e6d25b82b6e123d94cb93318de1fabded4bb1d3045447c2e9b0d3d0051c",
    "block_number": 4194
  }
]
```

File: code/go/0chain.net/smartcontract/storagesc/writemarker_eventdb.go
Comment (kushthedude) [inline code/go/0chain.net/smartcontract/storagesc/writemarker_eventdb.go]:
```suggestion
		return fmt.Errorf("failed to marshal writemarker: %v", err)
```

**Review by kushthedude — DISMISSED** [review]
Very minor change, but PR looks awesome to me! Will wait for @Sriep review

**Comment by ssardana96** [comment]
PR merge blocked on #857

**Review by Sriep — DISMISSED** [review]
Looks good after merging in the current staging. 

Since there is no backward compatibility issues can be merged. However, needs integration testing with explorer.

These changes need to be integration tested with upcoming explorer changes. Check with @shaktals.

**Review by kushthedude — APPROVED** [review]

**Pull request merged**
